### PR TITLE
Use react-motion to unmount Portal children in Expander

### LINKED ISSUE #421: `Portal`s inside of `ExpanderPanel` persist through `Expander` collapse
https://appnexus.slack.com/archives/anx-react/p1468343181000019

### LINKED ISSUE #445: Sidebar overflow when collapsed
A gif is worth a 10k words.

bug

### PR DESCRIPTION
**Pull request by ogupte**
## PR Checklist

- Manually tested across supported browsers
  - [x] Chrome
  - [x] Firefox
  - [x] Safari
  - [x] IE11 (Win 7)
  - [x] Edge (Win 10)
- [x] Unit tests written (`common` at minimum)
- [x] PR has one of the `semver-` labels
- [x] Two core team engineer approvals
- ~~One core team UX approval~~

elements within a narrow range expander height values.

Fixes #421
Fixes #445

### CODE REVIEW

**Comment by jondlm** [comment]
Sadly there's a third place where this will need to be implmented:

```
src/components/VerticalListMenu/VerticalListMenu.less
93:                     .expandable-animation-collapsed();
```

File: src/components/Expander/Expander.jsx [hunk 1]
```diff
@@ -102,6 +106,44 @@ const Expander = createClass({
...
+		this._labelKey = 0;
+	},
+
+	componentDidMount() {
+		_.delay(() => {
+			this.setState({
+				maxHeight: this.Refs.contentInner.offsetHeight,
+			});
+			this.isAnimated = true;
+		}, 32);
```
Comment (jondlm) [inline src/components/Expander/Expander.jsx]: What's the purpose of this constant? Could probably move it to a `const` to make a little clearer

**Comment by jondlm** [comment]
Also noticed an issue with FF
doh

**Comment by jondlm** [comment]
Could you upload the latest to docspot so I can take a peek?

File: src/components/Collapsible/Collapsible.jsx [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+		};
+	},
+
+	componentWillMount() {
+		this.Refs = {};
+		this.isAnimated = false;
+	},
+
+	componentDidMount() {
+		_.delay(() => {
```
Comment (sodiumjoe) [inline src/components/Collapsible/Collapsible.jsx]: what's this for?

File: src/components/Collapsible/Collapsible.jsx [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+		return (
+			<Motion
+				style={this.isAnimated ? {
+					height: (isExpanded && !_.isNull(maxHeight) ? spring(maxHeight, QUICK_SLIDE_MOTION) : spring(0, QUICK_SLIDE_MOTION)),
+				} : {
+					height: (isExpanded && !_.isNull(maxHeight) ? maxHeight : 0),
+				}}
+			>
+				{tween => (
+					React.createElement(rootType, {
```
Comment (sodiumjoe) [inline src/components/Collapsible/Collapsible.jsx]: why not just use jsx?

File: src/components/ExpanderPanel/ExpanderPanel.jsx [hunk 4]
```diff
@@ -138,13 +139,13 @@ const ExpanderPanel = createClass({
 					<span {...headerChildProps} />
 				</Panel.Header>
 
-				<div className={cx('&-content', {
+				<Collapsible isExpanded={isExpanded} className={cx('&-content', {
 					'&-content-is-expanded': isExpanded,
```
Comment (sodiumjoe) [inline src/components/ExpanderPanel/ExpanderPanel.jsx]: keeping this for backwards compat?

File: src/components/VerticalListMenu/VerticalListMenu.jsx [hunk 5]
```diff
@@ -199,11 +200,14 @@ const VerticalListMenu = createClass({
 							: null}
 							</div>
 
-							<div className={cx('&-Item-nested-list', {
-								'&-Item-nested-list-is-expanded': actualIsExpanded,
-							})}>
-								{listChildren}
-							</div>
+							{!_.isEmpty(listChildren) ? (
```
Comment (sodiumjoe) [inline src/components/VerticalListMenu/VerticalListMenu.jsx]: why do you need this?

**Comment by jondlm** [comment]
Just manually tested across our supported browsers, works great!

File: src/components/Collapsible/Collapsible.jsx (hunk 3 above)
Comment (jondlm) [inline src/components/Collapsible/Collapsible.jsx]: That's how I did it in [context menu](https://github.com/appnexus/lucid/blob/v1.10.0/src/components/ContextMenu/ContextMenu.jsx#L374)

File: src/util/generic-tests.jsx [hunk 6]
```diff
@@ -4,13 +4,15 @@ import { mount, shallow } from 'enzyme';
...
 import * as lucid from '../index';
+import { Motion } from 'react-motion';
 
 // Common tests for all our components
 export function common(Component, {
 	getDefaultProps = _.constant({}),
 	selector = null,
 	exemptFunctionProps = [],
 	exemptChildComponents = [],
+	selectRoot = _.identity,
```
Comment (jondlm) [inline src/util/generic-tests.jsx]: Cool beans

**Pull request merged**